admin/preprints/views.py

```diff
@@ -324,9 +325,12 @@ class WithdrawalRequestMixin(PermissionRequiredMixin):
     permission_required = 'osf.change_preprintrequest'
 
     def get_object(self):
+        guid = self.kwargs['guid']
+        if VersionedGuidMixin.GUID_VERSION_DELIMITER in guid:
+            guid = guid.split(VersionedGuidMixin.GUID_VERSION_DELIMITER)[0]
         return PreprintRequest.objects.filter(
             request_type='withdrawal',
-            target__guids___id=self.kwargs['guid'],
+            target__guids___id=guid,
         ).first()
```

Comment:
The design of the versioned GUID is: if version exists, find the object with the version instead of finding the object with the base GUID.

For this case, we need to change how we query the object in filter `.filter()`.

Take a look at model `PreprintRequest`, you will see `target` is a foreign key to the preprint object. Here we need to different cases:

* When `guid` is non versioned string: filter with `target__guids___id=guid` works
* When `guid` is a versioned string. You should filter on both the base guide and the version.